src/v2/blobs.rs

```diff
@@ -49,11 +49,13 @@ impl Client {
             }
         };
 
-        let fres = self.build_reqwest(reqwest::async::Client::new().get(url))
+        let fres_digest = futures::future::result(ContentDigest::try_new(digest.to_string()));
```

Comment:
I fear this should be moved even earlier. This line does input validation for `digest`, but we already use `digest` for format+parse. It should logically precede that.

If the url-parsing becomes too unwieldy  because of that, you can consider refactoring into a chain like https://github.com/camallo/dkregistry-rs/blob/0d4317d0e9816568d974d94570a7b87d3230f20e/src/v2/mod.rs#L97-L105